vision/emergent_object/emergent_object_processing.py

```diff
@@ -0,0 +1,161 @@
+"""Picks the most likely emergent object out of the pool of candidates"""
+
+from nptyping import NDArray, Shape, Float64
+
+import numpy as np
+import geopy.distance
+
+from vision.common.bounding_box import BoundingBox, ObjectType, Vertices
+
+from vision.common.constants import Location, ODLC_Dict
+
+
+def pick_emergent_object(humanoids: list[BoundingBox], odlcs: ODLC_Dict) -> BoundingBox:
+    """
+    Evaluates all of the potential emergent objects to pick the most likely candidate.
+    The metrics that are used in the evaluation are, in order:
+
+    Distance to ODLC Standard Object: 50% of final score
+        The judges are more likely to be close to the ODLC Standard Object. Thus, candidates
+        that are farther away will be favored.
+
+    Distance to Emergent Object (in a DIFFERENT image): 25% of final score
+        This will be smaller for the emergent object - because we have detected it in the
+        same place twice. Judges will be walking around a lot more and will be detected in
+        different places at different times.
+
+    Bounding Box Area: 20% of final score:
+        The Emergent Object will be placed on the ground and (presumably) not standing up.
+        This means that the area of its bounding box should, in general, be larger than
+        that of the judges.
+
+    AI Confidence Level: 5% of final score:
+        Higher confidence levels should be favored
+
+    Parameters
+    ----------
+    humanoids: list[BoundingBox]
+        A list of all candidate emergent objects. Required attributes:
+            latitude, longitude, bounding_box_area, image_path, confidence
+    odlcs: ODLC_Dict
+        A dictionary containing the locations of all discovered ODLCs
+
+    Returns
+    -------
+    emergent_object: BoundingBox
+        The candidate that has scored the highest, and so is probably the actual emergent
+        object
+    """
+
+    # Weights for each metric
+    category_weights: NDArray[Shape[4], Float64] = np.array([0.5, 0.25, 0.20, 0.05])
+
+    evaluations_list: list[tuple[float, float, float, float]] = []
+
+    subject_humanoid: BoundingBox
+    for subject_humanoid in humanoids:
+        # Get the distance to the closest standard object
+        min_odlc_distance: float = float("inf")
+        odlc: Location
+        for odlc in odlcs.values():
+            odlc_distance: float = geopy.distance.geodesic(
+                (odlc["latitude"], odlc["longitude"]),
+                (subject_humanoid.attributes["latitude"], subject_humanoid.attributes["longitude"]),
+            ).feet
+
+            min_odlc_distance = min(odlc_distance, min_odlc_distance)
```

Comment:
Small nitpick, but this would be more efficient as an if statement. Right now, you are always reassigning `min_odlc_distance`, no matter if it is just getting the same value again. With an if statement, you can check which is smaller (which is the same as the `min`), and then only do the assignment if you have to.